crates/compiler/gen_dev/src/generic64/mod.rs

```diff
@@ -746,6 +770,40 @@ impl<
                     .load_to_general_reg(&mut self.buf, src2);
                 ASM::imul_reg64_reg64_reg64(&mut self.buf, dst_reg, src1_reg, src2_reg);
             }
+            Layout::Builtin(Builtin::Int(
+                IntWidth::U64 | IntWidth::U32 | IntWidth::U16 | IntWidth::U8,
+            )) => {
+                // TODO find a general way to do this
+                let rax = CC::GENERAL_RETURN_REGS[0];
+                let rdx = CC::GENERAL_RETURN_REGS[1];
```

Comment:
is there a generic way to do this? in general, aren't we mixing x86_64 details here with the generic part of the dev backend?

Reply:
Oh, sorry, this needs to be dealt with inside of `ASM::umul_reg64_reg64_reg64`. It will be different between x86 and arm. You can pass the storage manager into umul. Some ops just have to be slightly special.